src/common/datetime/format_date.ts

```diff
@@ -32,15 +32,55 @@ const formatDateMem = memoizeOne(
 
 // 10/08/2021
 export const formatDateNumeric = (dateObj: Date, locale: FrontendLocaleData) =>
-  formatDateNumericMem(locale).format(dateObj);
+  formatDateNumericMem(locale, dateObj);
 
 const formatDateNumericMem = memoizeOne(
-  (locale: FrontendLocaleData) =>
-    new Intl.DateTimeFormat(locale.language, {
+  (locale: FrontendLocaleData, dateObj: Date) => {
+    let localeString: string | undefined = locale.language;
+
+    if (locale.date_format === DateFormat.system) {
+      localeString = undefined;
+    }
+
+    let result = new Intl.DateTimeFormat(localeString, {
       year: "numeric",
       month: "numeric",
       day: "numeric",
-    })
+    }).format(dateObj);
```

Comment:
If system or use language we can just return here, no need to do all the stuff below